src/devices/bus/coco/meb_intrf.cpp

```diff
@@ -0,0 +1,206 @@
+
+
+//-------------------------------------------------
+//  distomeb_slot_device - constructor
+//-------------------------------------------------
+distomeb_slot_device::distomeb_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: device_t(mconfig, DISTOMEB_SLOT, tag, owner, clock)
+	, device_single_card_slot_interface<device_distomeb_interface>(mconfig, *this)
+	, m_cart_callback(*this)
+	, m_cart(nullptr)
+{
+}
+
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void distomeb_slot_device::device_start()
+{
+	m_cart = get_card_device();
+}
+
+
+//-------------------------------------------------
+//  meb_read
+//-------------------------------------------------
+
+u8 distomeb_slot_device::meb_read(offs_t offset)
+{
+	u8 result = 0x00;
+	if (m_cart)
+		result = m_cart->meb_read(offset);
+	return result;
+}
+
+
+//-------------------------------------------------
+//  meb_write
+//-------------------------------------------------
+
+
+void distomeb_slot_device::meb_write(offs_t offset, u8 data)
+{
+	if (m_cart)
+		m_cart->meb_write(offset, data);
+}
+
+//-------------------------------------------------
+//  set_cart_line
+//-------------------------------------------------
+
+void distomeb_slot_device::set_cart_line(int value)
+{
+	m_cart_line = value;
+	if (!(m_cart_callback.isnull()))
+		m_cart_callback(value);
+}
+
+
+//-------------------------------------------------
+//  get_cart_line
+//-------------------------------------------------
+
+int distomeb_slot_device::get_cart_line() const
+{
+	return m_cart_line;
+}
+
+
+//**************************************************************************
+//  DEVICE DISTO MEB INTERFACE - Implemented by devices that plug into
+//  Disto Mini Expansion Bus
+//**************************************************************************
+
+template class device_finder<device_distomeb_interface, false>;
+template class device_finder<device_distomeb_interface, true>;
+
+
+//-------------------------------------------------
+//  device_distomeb_interface - constructor
+//-------------------------------------------------
+
+device_distomeb_interface::device_distomeb_interface(const machine_config &mconfig, device_t &device)
+	: device_interface(device, "distomeb")
+	, m_owning_slot(nullptr)
+{
+}
+
+
+//-------------------------------------------------
+//  ~device_distomeb_interface - destructor
+//-------------------------------------------------
+
+device_distomeb_interface::~device_distomeb_interface()
+{
+}
+
+
+//-------------------------------------------------
+//  interface_config_complete
+//-------------------------------------------------
+
+void device_distomeb_interface::interface_config_complete()
+{
+	m_owning_slot = dynamic_cast<distomeb_slot_device *>(device().owner());
+}
```

Comment:
You may as well make `m_owning_slot` a `const` pointer and initialise it on construction – the owner can’t change after construction.

Reply:
I switch to initializing in the constructor (like all the other [non CoCo] slots systems do). But when I add the const modifier I get a compile error I do not know how to resolve:

`Compiling src/devices/bus/coco/meb_intrf.cpp...
../../../../../src/devices/bus/coco/meb_intrf.cpp: In member function 'void device_distomeb_interface::set_cart_value(int)':
../../../../../src/devices/bus/coco/meb_intrf.cpp:176:37: error: passing 'const distomeb_slot_device' as 'this' argument discards qualifiers [-fpermissive]
  176 |         m_owning_slot->set_cart_line(value);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~`

Reply:
Not a pointer to a `const` object, a `const` pointer.